### ISSUE TITLE
Invalid Request

### ISSUE THREAD

**Issue by joe455** [aws/amazon-cognito-auth-js#55]
I am using cognito auth for custom UI facebook login in local environment. It is prompting for facebook credentials but on redirecting it is showing invalid request.Any Idea on this?

**Comment by jonasao** [comment]
Have you entered the correct redirect value in your AWS Cognito user pool?

**Comment by yuntuowang** [comment]
@jonasao is correct. Can you check your redirect URL setting under app client settings are consistent with the one you configured in your project? @joe455

**Comment by joe455** [comment]
my configuration in project is something like this.

`const authData = {
  ClientId: '2hjcqkf1l6hvv6h2drhb2a02cn',
  AppWebDomain : 'localhost:4200',
  TokenScopesArray : ['profile', 'email'],
  //TokenScopesArray : ['profile', 'email', 'openid', 'aws.cognito.signin.user.admin', 'phone'],
  RedirectUriSignIn : 'http://localhost:4200/home',
  RedirectUriSignOut : 'http://localhost:4200/home'
};`

and cognito seettings redirect url is 'http://localhost:4200/home'

Any mistake in this?

**Comment by joe455** [comment]
Now its redirecting to **https**://localhost:4200/oauth2/authorize?redirect_uri=http%3A%2F%2Flocalhost%3A4200%2Fhome&response_type=token&client_id=***&state=**&scope=profile%20email
 when I change it to https to http I am getting angular error(For obvious reason though)

**Error: Cannot match any routes. URL Segment: 'oauth2/authorize'**

@jonasao @yuntuowang I there is any problem with my domain name?

**Comment by yuntuowang** [comment]
Hi @joe455 , currently we don't support http schema. 
Also, for "AppWebDomain" field, it is set in your Cognito console "Domain name" tab. For example: "etgbnhwr.auth.us-east-1.amazoncognito.com".
The example hosted UI endpoint: https://****.auth.us-east-1.amazoncognito.com/login?response_type=token&client_id=*******&redirect_uri=https://www.amazon.com

**Comment by joe455** [comment]
Is there any possibility to achieve same thing with custom UI?

**Comment by yuntuowang** [comment]
@joe455 Do you mean you want to use your own UI but not our hosted UI page?  Actually our hosted UI is customizable. You can change color and a lot more with the parts of the page.

**Comment by joe455** [comment]
@yuntuowang Yes, I know the possibilities with the hosted UI. I tried everything over there.
Although my client needs custom UI(App with angular4). Is there any possibility for this?
Any SDK/documentation or anything related for this?

**Comment by yuntuowang** [comment]
Hi @joe455 , I totally understand your need for your clients. Currently we don't support using your own UI. However, I will mark this as a feature request and discuss during our meetings. I will post on this issue once we have any new status. Thanks a lot!

**Comment by joe455** [comment]
Thanks @yuntuowang 
Looking forward to that feature.Thanks for your support !!!!

**Comment by yuntuowang** [comment]
@joe455 You are very welcome!!! Let me know if you have any other questions.

**Comment by joe455** [comment]
I am able to save the federated users into userpool without using hosted UI(By providing IdentityProvider attribute). Thanks alot 👍

**Comment by yuntuowang** [comment]
Hi @joe455, you are very welcome!! Glad it works!!

**Comment by Jun711** [comment]
@joe455 Hi Joe, just wonder if you could share a gist of your working copy. thanks

**Comment by engharb** [comment]
@joe455 @yuntuowang I have the same problem. When using Hosted UI in my Web app the FB login process works perfect and then the identity provider instance is linked/merged in user pool. But that's does not work in Mobile Apps. Therefore I am using Ionic Facebook to login (not Hosted UI), my question is @joe455  how did you succeeded in using IdentityProvider attr to link the fb_identityId in user pool?

regards,

**Comment by guilhermedecampo** [comment]
@engharb did you find how to configure FB login to link/merge in user pool? 
I have a custom implementation too and it's being super hard to configure and find information about it.

**Comment by gitsome** [comment]
I see a solution for adding federated user to user pool was found. @yuntuowang was the feature of supporting custom UI for this ever added to the Amplify roadmap? We are federating users via a social idp and need User Pool support but would really love more control over the user experience during the login process.

**Comment by yuntuowang** [comment]
Hi @gitsome, adding support of custom UI will be a feature request from Cognito service side. At this point, I cannot comment on the delivery time.
